marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -140,55 +141,58 @@ getMarloweConstructors PartyType =
 -- | Creates a String consisting of the data constructor name followed by argument holes
 --   e.g. "When [ ?case_10 ] ?timeout_11 ?contract_12"
 constructMarloweType :: String -> MarloweHole -> Map String (Array Argument) -> String
-constructMarloweType constructorName (MarloweHole { name, marloweType, start }) m = case Map.lookup constructorName m of
+constructMarloweType constructorName (MarloweHole { name, marloweType, line, col }) m = case Map.lookup constructorName m of
   Nothing -> ""
   Just [] -> constructorName
-  Just vs -> "(" <> constructorName <> " " <> intercalate " " (mapWithIndex (showArgument start) vs) <> ")"
+  Just vs -> "(" <> constructorName <> " " <> intercalate " " (mapWithIndex (showArgument col) vs) <> ")"
   where
   showArgument startPos i (ArrayArg arg) = "[ ?" <> arg <> "_" <> show (startPos + i) <> " ]"
 
   showArgument startPos i (DataArg arg) = "?" <> arg <> "_" <> show (startPos + i)
 
   showArgument _ _ NewtypeArg = ""
 
+mkHole :: forall a. String -> { row :: Pos, column :: Pos } -> Term a
+mkHole name position = Hole name Proxy position
+
 data Term a
-  = Term a Pos Pos
-  | Hole String (Proxy a) Pos Pos
+  = Term a { row :: Pos, column :: Pos }
+  | Hole String (Proxy a) { row :: Pos, column :: Pos }
 
 derive instance genericTerm :: Generic (Term a) _
 
 instance eqTerm :: Eq a => Eq (Term a) where
   eq a b = genericEq a b
 
 instance showTerm :: Show a => Show (Term a) where
-  show (Term a _ _) = show a
-  show (Hole name _ _ _) = "?" <> name
+  show (Term a _) = show a
+  show (Hole name _ _) = "?" <> name
 
 instance prettyTerm :: Pretty a => Pretty (Term a) where
-  pretty (Term a _ _) = P.pretty a
-  pretty (Hole name _ _ _) = P.text $ "?" <> name
+  pretty (Term a _) = P.pretty a
+  pretty (Hole name _ _) = P.text $ "?" <> name
 
 instance hasArgsTerm :: Args a => Args (Term a) where
-  hasArgs (Term a _ _) = hasArgs a
+  hasArgs (Term a _) = hasArgs a
   hasArgs _ = false
-  hasNestedArgs (Term a _ _) = hasNestedArgs a
+  hasNestedArgs (Term a _) = hasNestedArgs a
   hasNestedArgs _ = false
 
 -- a concrete type for holes only
 data MarloweHole
   = MarloweHole
     { name :: String
     , marloweType :: MarloweType
-    , start :: Pos
-    , end :: Pos
+    , line :: Pos
```

Comment:
Nitpick, but throughout this you seem to be interchanging `line` and `row`. I think you should pick one and stick to it.

Actually, this might be easier if you started using the type alias you've got for row/column records more liberally.

Reply:
in the purescript code it should always be using `row` which is what Ace uses, I see now that `MarloweHole` is using the wrong terms, I will change them there. The grammar unfortunately uses `line` but Ive tried to keep it in there.